src/schedule/dma_multi_chan_domain.c

```diff
@@ -241,6 +241,10 @@ static bool dma_multi_chan_domain_is_pending(struct ll_schedule_domain *domain,
 			if (dma_domain->data[i][j].task != task)
 				continue;
 
+			/* execute callback if exists */
+			if (dmas[i].chan[j].irq_callback)
+				dmas[i].chan[j].irq_callback(&dmas[i].chan[j]);
+
```

Comment:
Is this the correct ordering? First run the callback, then clear the IRQ? Does this mean that certain actions which may cause a second DMA IRQ to come in the very brief time period between entering this function and then folllowing DMA_IRQ_CLEAR call shall be forbidden?

Reply:
@paulstelian97 It doesn't really matter, because it's all happening in critical section of low latency scheduler. You won't get any additional interrupt until all processing is done.

Reply:
Ah, I see.